### DISCUSSION TITLE
support for static generate?

### DISCUSSION THREAD

**Discussion by pixelpaulaus** [johannschopplich/nuxt-kql#19] [Q&A]
Hello,
Does this module support `npx nuxi generate` to create a static site?

**Comment by johannschopplich** [comment]
Absolutely, it does. I recommend the [Kirby Cacao Kit](https://github.com/johannschopplich/cacao-kit-frontend/), which already includes best practices for working with headless Kirby.

**Reply by pixelpaulaus** [reply]
thanks. I am testing it and trying to do it. I assume it is via the prefetch query. Which just queries 'site'. do I need to change that to query all pages? as it doesn't seem to pull in my '/blog/*' pages from kirby and pre-render them

**Reply by johannschopplich** [reply]
No, the prefetch query just prefetches any query you want at build-time. This is useful if you need some static data that doesn't need to be fetched at runtime.

For prerender settings, you have to follow the [Nuxt docs](https://nuxt.com/docs/getting-started/prerendering). You're probably missing the `routes` or `crawlLinks` option.

**Reply by pixelpaulaus** [reply]
that is what i was thinking. so there is no way to pull all the routes from the kirby at this stage and pre-render them. I need to manually code them into the routes. Or all pages need to be linked to for the crawler to pick them up.

**Reply by johannschopplich** [reply]
Yes, there is a way, which I use for some projects. You can fetch a page tree from Kirby in a custom (local) Nuxt module and then add these routes to the Nitro prerenderer.
